radix-engine/tests/resource/src/lib.rs

```diff
@@ -22,59 +24,60 @@ blueprint! {
+                .auth("take_from_vault", auth!(allow_all))
                 .divisibility(DIVISIBILITY_NONE)
                 .initial_supply(1);
             let token_address = ResourceBuilder::new_fungible()
                 .divisibility(divisibility)
                 .metadata("name", "TestToken")
-                .flags(MINTABLE | BURNABLE)
-                .badge(badge.resource_address(), MAY_MINT | MAY_BURN)
+                .auth("take_from_vault", auth!(allow_all))
+                .auth("mint", auth!(require(badge.resource_address())))
+                .auth("burn", auth!(require(badge.resource_address())))
                 .no_initial_supply();
             let tokens = badge.authorize(|| resource_manager!(token_address).mint(amount));
             (badge, tokens, token_address)
         }
 
         pub fn create_fungible_wrong_resource_flags_should_fail() -> ResourceAddress {
             let token_address = ResourceBuilder::new_fungible()
+                .auth("take_from_vault", auth!(allow_all))
                 .divisibility(DIVISIBILITY_MAXIMUM)
                 .metadata("name", "TestToken")
-                .flags(MAY_MINT | BURNABLE)
                 .no_initial_supply();
             token_address
         }
 
         pub fn create_fungible_wrong_mutable_flags_should_fail() -> ResourceAddress {
             let token_address = ResourceBuilder::new_fungible()
+                .auth("take_from_vault", auth!(allow_all))
                 .divisibility(DIVISIBILITY_MAXIMUM)
                 .metadata("name", "TestToken")
-                .flags(MINTABLE | BURNABLE)
-                .mutable_flags(MAY_MINT)
                 .no_initial_supply();
             token_address
         }
 
         pub fn create_fungible_wrong_resource_permissions_should_fail() -> (Bucket, ResourceAddress)
         {
             let badge = ResourceBuilder::new_fungible()
+                .auth("take_from_vault", auth!(allow_all))
                 .divisibility(DIVISIBILITY_NONE)
                 .initial_supply(1);
             let token_address = ResourceBuilder::new_fungible()
                 .divisibility(DIVISIBILITY_MAXIMUM)
                 .metadata("name", "TestToken")
-                .flags(MINTABLE | BURNABLE)
-                .badge(badge.resource_address(), MINTABLE | MAY_BURN)
+                .auth("take_from_vault", auth!(allow_all))
+                .auth("mint", auth!(require(badge.resource_address())))
+                .auth("burn", auth!(require(badge.resource_address())))
                 .no_initial_supply();
             (badge, token_address)
         }
 
-        pub fn query() -> (Bucket, HashMap<String, String>, u64, u64, Decimal) {
-            let (badge, resource_address) = Self::create_fungible();
-            let resource_manager = resource_manager!(resource_address);
+        pub fn query() -> (Bucket, HashMap<String, String>, Decimal) {
+            let (badge, resource_def_id) = Self::create_fungible();
```

Comment:
Can we rename these `resource_def_id`? :)

Reply:
oops